### PR TITLE
fix: enable errorlint in server directory (part 1)

### PR DESCRIPTION
**Pull request by mmorel-35**
This enable errorlint in packages server 

Related to #18576

### CODE REVIEW

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks

**Comment by ahrtr** [comment]
> > Please install the 'codecov app svg image' to ensure uploads and comments are reliably processed by Codecov.
> 
> Is there any intention to address this ?

Thank you for bringing this up. Where did you see the message? 

I'd like to defer to @henrybear327 and @ivanvc to comment on this. Thanks.

**Comment by ivanvc** [comment]
> I'd like to defer to @henrybear327 and @ivanvc to comment on this. Thanks.

This was briefly discussed in issue https://github.com/etcd-io/etcd/issues/18131#issuecomment-2149414835 / https://github.com/etcd-io/etcd/pull/18143. At that point, it worked fine, but that banner/warning later started appearing (see how it was before https://github.com/etcd-io/etcd/pull/18143).

Let's track it in the original issue (cc. @henrybear327). Thanks, @mmorel-35, for raising awareness.

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @mmorel-35 

I have added a comment https://github.com/etcd-io/etcd/issues/18131#issuecomment-2433182954 in relation to the codecov warning.

File: server/etcdmain/config.go
Comment (serathius) [inline server/etcdmain/config.go:119]: Think this has different semantics than checking `err == nil`. 
How this is an improvement?

**Comment by ivanvc** [comment]
> @ahrtr,
> 
> There is no suggested assignee here. Do you know why ? Who shall be assigned then ?

Hi @mmorel-35, given the size of the etcd team, I don't think we're currently using the assignee field in pull requests. Feel free to keep adding them or skip them :sweat_smile:. We have some maintainers who are on leave and others who are not tech leads. The tech leads are @ahrtr and @serathius :)

**Pull request closed without merging**
